From 1b2c0b2b21d6158f0a033def474f70af2129c84c Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Thu, 25 Mar 2021 12:20:06 +0100
Subject: [PATCH 072/117] Revert "bfq: expire other class if CLASS_RT is
 waiting"

This reverts commit eeb06cebb760348f49ea5ea13eb85ebe518e663d.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/bfq-iosched.c | 15 +++++----------
 block/bfq-iosched.h |  8 --------
 block/bfq-wf2q.c    | 12 ------------
 3 files changed, 5 insertions(+), 30 deletions(-)

diff --git a/block/bfq-iosched.c b/block/bfq-iosched.c
index acfeeaae8..c31663bc0 100644
--- a/block/bfq-iosched.c
+++ b/block/bfq-iosched.c
@@ -426,7 +426,6 @@ void bfq_schedule_dispatch(struct bfq_data *bfqd)
 	}
 }
 
-#define bfq_class_rt(bfqq)	((bfqq)->ioprio_class == IOPRIO_CLASS_RT)
 #define bfq_class_idle(bfqq)	((bfqq)->ioprio_class == IOPRIO_CLASS_IDLE)
 
 #define bfq_sample_valid(samples)	((samples) > 80)
@@ -4819,16 +4818,12 @@ static struct request *bfq_dispatch_rq_from_bfqq(struct bfq_data *bfqd,
 	/*
 	 * Expire bfqq, pretending that its budget expired, if bfqq
 	 * belongs to CLASS_IDLE and other queues are waiting for
-	 * service, or if bfqq not belongs to CLASS_RT and CLASS_RT
-	 * is waiting for service.
+	 * service.
 	 */
-#ifdef CONFIG_BFQ_GROUP_IOSCHED
-	if ((bfq_tot_busy_queues(bfqd) > 1 && bfq_class_idle(bfqq)) ||
-	    (!bfq_class_rt(bfqq) && bfqd->busy_groups[0]))
-#else
-	if (bfq_tot_busy_queues(bfqd) > 1 && bfq_class_idle(bfqq))
-#endif
-		bfq_bfqq_expire(bfqd, bfqq, false, BFQQE_BUDGET_EXHAUSTED);
+	if (!(bfq_tot_busy_queues(bfqd) > 1 && bfq_class_idle(bfqq)))
+		goto return_rq;
+
+	bfq_bfqq_expire(bfqd, bfqq, false, BFQQE_BUDGET_EXHAUSTED);
 
 return_rq:
 	return rq;
diff --git a/block/bfq-iosched.h b/block/bfq-iosched.h
index cabd2adcf..fc4fd0258 100644
--- a/block/bfq-iosched.h
+++ b/block/bfq-iosched.h
@@ -543,14 +543,6 @@ struct bfq_data {
 	 */
 	unsigned int num_groups_with_pending_reqs;
 
-#ifdef CONFIG_BFQ_GROUP_IOSCHED
-	/*
-	 * Per-class (RT, BE, IDLE) number of bfq_groups waiting for
-	 * service.
-	 */
-	unsigned int busy_groups[3];
-#endif
-
 	/*
 	 * Per-class (RT, BE, IDLE) number of bfq_queues containing
 	 * requests (including the queue in service, even if it is
diff --git a/block/bfq-wf2q.c b/block/bfq-wf2q.c
index aa9f965c3..bc9150d0c 100644
--- a/block/bfq-wf2q.c
+++ b/block/bfq-wf2q.c
@@ -1032,14 +1032,11 @@ static void __bfq_activate_entity(struct bfq_entity *entity,
 	if (!bfq_entity_to_bfqq(entity)) { /* bfq_group */
 		struct bfq_group *bfqg = bfq_entity_to_bfqg(entity);
 		struct bfq_data *bfqd = bfqg->bfqd;
-		int idx = bfq_class_idx(entity);
 
 		if (!entity->in_groups_with_pending_reqs) {
 			entity->in_groups_with_pending_reqs = true;
 			bfqd->num_groups_with_pending_reqs++;
 		}
-
-		bfqd->busy_groups[idx]++;
 	}
 #endif
 
@@ -1191,9 +1188,6 @@ bool __bfq_deactivate_entity(struct bfq_entity *entity, bool ins_into_idle_tree)
 {
 	struct bfq_sched_data *sd = entity->sched_data;
 	struct bfq_service_tree *st;
-#ifdef CONFIG_BFQ_GROUP_IOSCHED
-	struct bfq_group *bfqg;
-#endif
 	int idx = bfq_class_idx(entity);
 	bool is_in_service;
 
@@ -1235,12 +1229,6 @@ bool __bfq_deactivate_entity(struct bfq_entity *entity, bool ins_into_idle_tree)
 		bfq_idle_insert(st, entity);
 
 	sd->bfq_class_last_service[idx] = jiffies;
-#ifdef CONFIG_BFQ_GROUP_IOSCHED
-	bfqg = bfq_entity_to_bfqg(entity);
-	if (bfqg)
-		bfqg->bfqd->busy_groups[idx]--;
-#endif
-
 	return true;
 }
 
-- 
2.31.1.305.gd1b10fc6d8

